test/NuGet.Clients.Tests/NuGet.CommandLine.Test/Util.cs

```diff
@@ -1237,5 +1237,195 @@ public static void TestCommandInvalidArguments(string command)
+            <SpecificVersion>True</SpecificVersion>
+        </Reference>");
+            }
+
+            projContent.AppendLine(
+@"        </ItemGroup>
+            <ItemGroup>
+            <None Include=""packages.config"" />
+            </ItemGroup>
+            <Import Project=""$(NanoFrameworkProjectSystemPath)NFProjectSystem.CSharp.targets"" Condition=""Exists('$(NanoFrameworkProjectSystemPath)NFProjectSystem.CSharp.targets')"" />
+            <ProjectExtensions>
+            <ProjectCapabilities>
+                <ProjectConfigurationsDeclaredAsItems />
+            </ProjectCapabilities>
+            </ProjectExtensions>
+        </Project>".Replace("$NAME$", projectName));
+
+            return projContent.ToString();
+        }
+
+        /// <summary>
+        /// Create a simple package with a lib folder for .NET nanoFramework
+        /// The package will be removed from the machine cache upon creation
+        /// </summary>
+        public static ZipPackage CreateNFPackage(string repositoryPath, string id, string version)
+        {
+            PackageBuilder package = Util.CreateTestPackageBuilder(id, version);
+            IPackageFile libFile = Util.CreatePackageFile("lib/a.dll", "a");
+            package.Files.Add(libFile);
+
+            string path = Util.CreateTestPackage(package, repositoryPath);
+
+            var zipPackage = new ZipPackage(path);
+
+            return zipPackage;
+        }
+
+        /// <summary>
+        /// Create a Solution file with .NET nanoFramework projects in it.
+        /// </summary>
+        /// <param name="projectList">List of .NET nanoFramework projects to add to the solution</param>
+        /// <returns>Content of the Solution file</returns>
+        public static string CreateNFSolutionFileContent(List<string> projectList)
+        {
+            var slnContent = new StringBuilder();
+
+            slnContent.AppendLine("Microsoft Visual Studio Solution File, Format Version 12.00");
+            slnContent.AppendLine("# Visual Studio Version 16");
+            slnContent.AppendLine("VisualStudioVersion = 16.0.31005.135");
+
+            foreach (string project in projectList)
+            {
+                slnContent.AppendLine($"Project(\"{{11A8DD76-328B-46DF-9F39-F559912D0360}}\") = \"{project}\", \"{project}\\{project}.nfproj\", \"{{{Guid.NewGuid().ToString()}}}");
+                slnContent.AppendLine("EndProject");
+            }
+
+            return slnContent.ToString();
+        }
+
+        /// <summary>
+        /// Creates a test package for .NET nanoFramework.
+        /// </summary>
+        /// <param name="packageId">Package ID</param>
+        /// <param name="version">Package version</param>
+        /// <param name="path">Path where the package will be saved</param>
+        /// <param name="contentFiles">Content file(s) to be added to the package</param>
+        /// <returns></returns>
+        public static string CreateNFTestPackage(
```

Comment:
Because we don't have a TFM, our DLLs have to be placed inside the lib folder. I couldn't find a nice way to do it with the existing methods. If there is one, please point me to it.

Reply:
@nkolev92 
Do you have any recommendation here?

Reply:
Look into [SimpleTestPackageContext](https://github.com/NuGet/NuGet.Client/blob/dev/test/TestUtilities/Test.Utility/SimpleTestSetup/SimpleTestPackageContext.cs) and then do `Files.Clear()` and add a few file paths, https://github.com/NuGet/NuGet.Client/blob/c6ca6ebcfe3fdad5815f19669c96431b53055b3b/test/TestUtilities/Test.Utility/SimpleTestSetup/SimpleTestPackageContext.cs#L52. 

If you search for references of Files, you'll likely find many tests already do that.

Reply:
@josesimoes 
I assume you're going to remove this one since it's not referenced anymore.